hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieBaseFileGroupRecordBuffer.java

```diff
@@ -115,9 +114,16 @@ public HoodieBaseFileGroupRecordBuffer(HoodieReaderContext<T> readerContext,
     } else {
       this.payloadClass = Option.empty();
     }
-    this.orderingFieldName = Option.ofNullable(ConfigUtils.getOrderingField(props)).orElseGet(() -> hoodieTableMetaClient.getTableConfig().getPreCombineField());
-    this.orderingFieldTypeOpt = recordMergeMode == RecordMergeMode.COMMIT_TIME_ORDERING ? Option.empty() : AvroSchemaUtils.findNestedFieldType(readerSchema, this.orderingFieldName);
-    this.orderingFieldDefault = orderingFieldTypeOpt.map(type -> readerContext.castValue(0, type)).orElse(0);
+    this.orderingFieldName = recordMergeMode == RecordMergeMode.COMMIT_TIME_ORDERING
```

Comment:
are we setting the ordering field to empty for regular ingestion writes itself? 
But how about backwards compatability?
if someone was using overwrite w/ latest payload in 0.15.0, but has ordering config set, we are forcing them to change to EVENT_TIME_ORDERING. but its a table property right?

can you help me understand

Reply:
We discussed that if the table version is 6 (Hudi 0.14/0.15 release), from the `HoodieTableConfig`, it should serve `EVENT_TIME_ORDERING` as the merge mode if the payload class is `OverwriteWithLatestAvroPayload` with precombine field in the table config, so that there is no behavior change of reading the table of version 6 with Hudi 1.x in this case.

Reply:
btw, can we check if this fix will take care of https://issues.apache.org/jira/browse/HUDI-8851

Reply:
Also note that, even prior to this PR, we're not using the ordering field if the merge mode is `COMMIT_TIME_ORDERING`.  The change here just to make the assumption tight and not cause any confusion.

Reply:
HUDI-8851 will be fixed by HUDI-8551 by handling precombine at the writer side.